### PR TITLE
public api

### PR DESCRIPTION
**Pull request by qibobo**

### CODE REVIEW

File: api/lib/config/setting.js [hunk 1]
```diff
@@ -65,6 +67,16 @@ module.exports = function(settingsObj) {
...
+    if (isMissing(settings.publicPort)){
+      return{valid:false, message:"publicPort is required"}
+    }
+    if (!isNumber(settings.publicPort)) {
+      return {valid:false,message:"publicPort must be a number"};
+    }
+    if (settings.publicPort < 1 || settings.publicPort > 65535) {
+      return {valid:false,message:"value of publicPort must between 1 and 65535"};
+    }
+
```
Comment (cdlliuy) [inline api/lib/config/setting.js]: how about add a check for public port != internal port ?

File: api/app.js [hunk 2]
```diff
@@ -77,20 +101,35 @@ module.exports = function(configFilePath) {
     });
   }
 
+  var publicServer;
+  if(settings.publicTls){
+    publicServer = https.createServer(publicOptions, app).listen(publicPort || 3002, function() {
```
Comment (cdlliuy) [inline api/app.js]: how about using different default port?

**Comment by qibobo** [comment]
@cdlliuy fixed.

**Comment by cdlliuy** [comment]
LGTM

File: api/lib/config/setting.js [hunk 3]
```diff
@@ -65,6 +67,18 @@ module.exports = function(settingsObj) {
...
     }
 
+    if (isMissing(settings.publicPort)){
+      return{valid:false, message:"publicPort is required"}
+    }
+    if (!isNumber(settings.publicPort)) {
+      return {valid:false,message:"publicPort must be a number"};
+    }
+    if (settings.publicPort < 1 || settings.publicPort > 65535) {
+      return {valid:false,message:"value of publicPort must between 1 and 65535"};
```
Comment (boyang9527) [inline api/lib/config/setting.js]: must -> must be

**Comment by boyang9527** [comment]
this PR does not include the permission check with the oAuth token... I am assuming that one will be added later.

**Comment by qibobo** [comment]
@boyang9527  fixed.

**Comment by qibobo** [comment]
@boyang9527  will have another pr for oauth.

**Comment by boyang9527** [comment]
LGTM

**Pull request merged**